sql: TestExperimentalRelocateNonVoters failed

sql.TestExperimentalRelocateNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14476594?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14476594?buildTab=artifacts#/) on master @ [f796c91258c2f365cdbb5ab081583a304ed96200](https://github.com/cockroachdb/cockroach/commits/f796c91258c2f365cdbb5ab081583a304ed96200):


```
=== RUN   TestExperimentalRelocateNonVoters
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0877ec60cc85705dc8073b5fa4b3cfc9/logTestExperimentalRelocateNonVoters3474815573
    test_log_scope.go:81: use -show-logs to present logs inline
    multitenant_admin_function_test.go:1007: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0877ec60cc85705dc8073b5fa4b3cfc9/logTestExperimentalRelocateNonVoters3474815573
--- FAIL: TestExperimentalRelocateNonVoters (12.02s)
=== RUN   TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
    multitenant_admin_function_test.go:380: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:380
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            				github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:42
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:377
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:996
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:346
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:971
        	Error:      	Received unexpected error:
        	            	pq: while carrying out changes [{ADD_NON_VOTER n3,s3} {REMOVE_NON_VOTER n5,s5}]: change replicas of r68 failed: descriptor changed: [expected] r68:/Tenant/1{0-1} [(n1,s1):1, (n4,s4):2, (n2,s2):3, (n5,s5):4NON_VOTER, (n3,s3):5NON_VOTER, next=6, gen=8, sticky=1710961420.689164218,0] != [actual] r68:/Tenant/10{-/Table/11} [(n1,s1):1, (n4,s4):2, (n2,s2):3, (n5,s5):4NON_VOTER, (n3,s3):5NON_VOTER, next=6, gen=9, sticky=1710961420.689164218,0]
        	Test:       	TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
        	Messages:   	tenant=secondary query=``ALTER TABLE t EXPERIMENTAL_RELOCATE NONVOTERS VALUES (ARRAY[3], 1);`` leaseholder=1 replicas=[1 2 4 5] voting_replicas=[1 2 4] non_voting_replicas=[5] fromReplica=5 toReplica=3
    --- FAIL: TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS (11.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExperimentalRelocateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestExperimentalRelocateNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14496279?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14496279?buildTab=artifacts#/) on master @ [fc6f8741a971b05a78639bba78a5c8982c488532](https://github.com/cockroachdb/cockroach/commits/fc6f8741a971b05a78639bba78a5c8982c488532):


```
=== RUN   TestExperimentalRelocateNonVoters
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0877ec60cc85705dc8073b5fa4b3cfc9/logTestExperimentalRelocateNonVoters2914702681
    test_log_scope.go:81: use -show-logs to present logs inline
    multitenant_admin_function_test.go:1007: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0877ec60cc85705dc8073b5fa4b3cfc9/logTestExperimentalRelocateNonVoters2914702681
--- FAIL: TestExperimentalRelocateNonVoters (12.39s)
=== RUN   TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
    multitenant_admin_function_test.go:380: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:380
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            				github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:42
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:377
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:996
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:346
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:971
        	Error:      	Received unexpected error:
        	            	pq: while carrying out changes [{ADD_NON_VOTER n4,s4} {REMOVE_NON_VOTER n3,s3}]: change replicas of r78 failed: descriptor changed: [expected] r78:/Tenant/1{0/Table/28-1} [(n1,s1):1, (n2,s2):2, (n5,s5):3, (n3,s3):5NON_VOTER, next=6, gen=15] != [actual] r78:/Tenant/1{0/Table/28-1} [(n1,s1):1, (n2,s2):2, (n5,s5):3, (n3,s3):5NON_VOTER, next=7, gen=17]
        	Test:       	TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
        	Messages:   	tenant=secondary query=``ALTER TABLE t EXPERIMENTAL_RELOCATE NONVOTERS VALUES (ARRAY[4], 1);`` leaseholder=1 replicas=[1 2 3 5] voting_replicas=[1 2 5] non_voting_replicas=[3] fromReplica=3 toReplica=4
    --- FAIL: TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS (12.37s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExperimentalRelocateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/kv/kvserver/helpers_test.go
+++ b/pkg/kv/kvserver/helpers_test.go
@@ -174,11 +174,6 @@ func (s *Store) SetReplicaGCQueueActive(active bool) {
 	s.setReplicaGCQueueActive(active)
 }
 
-// SetSplitQueueActive enables or disables the split queue.
-func (s *Store) SetSplitQueueActive(active bool) {
-	s.setSplitQueueActive(active)
-}
-
 // SetMergeQueueActive enables or disables the merge queue.
 func (s *Store) SetMergeQueueActive(active bool) {
 	s.setMergeQueueActive(active)

--- a/pkg/kv/kvserver/queue_helpers_testutil.go
+++ b/pkg/kv/kvserver/queue_helpers_testutil.go
@@ -129,7 +129,7 @@ func (s *Store) setReplicaGCQueueActive(active bool) {
 func (s *Store) SetReplicateQueueActive(active bool) {
 	s.replicateQueue.SetDisabled(!active)
 }
-func (s *Store) setSplitQueueActive(active bool) {
+func (s *Store) SetSplitQueueActive(active bool) {
 	s.splitQueue.SetDisabled(!active)
 }
 func (s *Store) setTimeSeriesMaintenanceQueueActive(active bool) {

--- a/pkg/kv/kvserver/store.go
+++ b/pkg/kv/kvserver/store.go
@@ -1699,7 +1699,7 @@ func NewStore(
 		s.SetReplicateQueueActive(false)
 	}
 	if cfg.TestingKnobs.DisableSplitQueue {
-		s.setSplitQueueActive(false)
+		s.SetSplitQueueActive(false)
 	}
 	if cfg.TestingKnobs.DisableTimeSeriesMaintenanceQueue {
 		s.setTimeSeriesMaintenanceQueueActive(false)

--- a/pkg/sql/multitenant_admin_function_test.go
+++ b/pkg/sql/multitenant_admin_function_test.go
@@ -737,6 +737,7 @@ func TestRelocateVoters(t *testing.T) {
 					err = testCluster.WaitForFullReplication()
 					require.NoErrorf(t, err, message)
 					testCluster.ToggleReplicateQueues(false)
+					testCluster.ToggleSplitQueues(false)
 					replicaState := getReplicaState(
 						t,
 						ctx,
@@ -816,6 +817,7 @@ func TestExperimentalRelocateVoters(t *testing.T) {
 					err = testCluster.WaitForFullReplication()
 					require.NoErrorf(t, err, message)
 					testCluster.ToggleReplicateQueues(false)
+					testCluster.ToggleSplitQueues(false)
 					replicaState := getReplicaState(
 						t,
 						ctx,
@@ -908,6 +910,7 @@ func TestRelocateNonVoters(t *testing.T) {
 					err = testCluster.WaitForFullReplication()
 					require.NoErrorf(t, err, message)
 					testCluster.ToggleReplicateQueues(false)
+					testCluster.ToggleSplitQueues(false)
 					replicaState := getReplicaState(
 						t,
 						ctx,
@@ -981,6 +984,7 @@ func TestExperimentalRelocateNonVoters(t *testing.T) {
 					err = testCluster.WaitForFullReplication()
 					require.NoErrorf(t, err, message)
 					testCluster.ToggleReplicateQueues(false)
+					testCluster.ToggleSplitQueues(false)
 					replicaState := getReplicaState(
 						t,
 						ctx,

--- a/pkg/testutils/serverutils/test_cluster_shim.go
+++ b/pkg/testutils/serverutils/test_cluster_shim.go
@@ -253,6 +253,10 @@ type TestClusterInterface interface {
 	// ToggleReplicateQueues activates or deactivates the replication queues on all
 	// the stores on all the nodes.
 	ToggleReplicateQueues(active bool)
+
+	// TogglesplitQueues activates or deactivates the split queues on all
+	// the stores on all the nodes.
+	ToggleSplitQueues(active bool)
 }
 
 // SplitPoint describes a split point that is passed to SplitTable.

--- a/pkg/testutils/testcluster/testcluster.go
+++ b/pkg/testutils/testcluster/testcluster.go
@@ -1613,6 +1613,16 @@ func (tc *TestCluster) ToggleReplicateQueues(active bool) {
 	}
 }
 
+// ToggleSplitQueues implements TestClusterInterface.
+func (tc *TestCluster) ToggleSplitQueues(active bool) {
+	for _, s := range tc.Servers {
+		_ = s.StorageLayer().GetStores().(*kvserver.Stores).VisitStores(func(store *kvserver.Store) error {
+			store.SetSplitQueueActive(active)
+			return nil
+		})
+	}
+}
+
 // ReadIntFromStores reads the current integer value at the given key
 // from all configured engines on un-stopped servers, filling in zeros
 // when the value is not found.

